src/components/lineup/Lineup.tsx

```diff
@@ -0,0 +1,295 @@
+          ? INITIAL_PLAYLISTS_MULTIPLER
+          : INITIAL_LOAD_TRACKS_MULTIPLIER
+      ),
+      loadMore: getItemCount(variant, TRACKS_AHEAD_MULTIPLIER)
+    }),
+    [variant]
+  )
+
+const styles = StyleSheet.create({
+  item: {
+    padding: 12,
+    paddingBottom: 0
+  }
+})
+
+/** `Lineup` encapsulates the logic for displaying a list of items such as Tracks (e.g. prefetching items
+ * displaying loading states, etc).
+ */
+export const Lineup = ({
+  actions,
+  count,
+  delineate,
+  isTrending,
+  leadingElementId,
+  limit,
+  lineup,
+  loadMore,
+  header,
+  playTrack,
+  pauseTrack,
+  rankIconCount = 0,
+  refresh,
+  refreshing,
+  showLeadingElementArtistPick = true,
+  variant
+}: LineupProps) => {
+  const dispatchWeb = useDispatchWeb()
+
+  const playing = useSelector(getPlaying)
+  const playingUid = useSelector(getPlayingUid)
+  const itemCounts = useItemCounts(variant)
+
+  // Item count based on the current page
+  const pageItemCount = useMemo(() => {
+    return itemCounts.initial + (lineup.page - 1) * itemCounts.loadMore
+  }, [itemCounts, lineup])
+
+  // Either the provided count or a default
+  const countOrDefault = useMemo(
+    () => (count !== undefined ? count : MAX_TILES_COUNT),
+    [count]
+  )
+
+  // When the onEndReachedThreshould is hit, potentially load
+  // more items
+  const onEndReached = useCallback(() => {
+    const { hasMore, page } = lineup
+
+    if (!hasMore) {
+      return
+    }
+
+    const lineupLength = lineup.entries.length
+    const offset = lineupLength + lineup.deleted
+
+    if (
+      (!limit || lineupLength !== limit) &&
+      loadMore &&
+      lineupLength < countOrDefault &&
+      (page === 0 || pageItemCount <= offset)
+    ) {
+      const itemLoadCount = itemCounts.initial + page * itemCounts.loadMore
+
+      dispatchWeb(actions.setPage(page + 1))
+
+      const limit =
+        Math.min(itemLoadCount, Math.max(countOrDefault, itemCounts.minimum)) -
+        offset
+
+      loadMore(offset, limit, page === 0)
+    }
+  }, [
+    actions,
+    countOrDefault,
+    dispatchWeb,
+    itemCounts,
+    limit,
+    lineup,
+    loadMore,
+    pageItemCount
+  ])
+
+  const togglePlay = useCallback(
+    (uid: UID, id: ID, source?: PlaybackSource) => {
+      if (uid !== playingUid || (uid === playingUid && !playing)) {
+        playTrack(uid)
+        track(
+          make({
+            eventName: Name.PLAYBACK_PLAY,
+            id: `${id}`,
+            source: source || PlaybackSource.TRACK_TILE
+          })
+        )
+      } else if (uid === playingUid && playing) {
+        pauseTrack()
+        track(
+          make({
+            eventName: Name.PLAYBACK_PAUSE,
+            id: `${id}`,
+            source: source || PlaybackSource.TRACK_TILE
+          })
+        )
+      }
+    },
+    [playTrack, pauseTrack, playing, playingUid]
+  )
+
+  const renderItem = ({ index, item }: { index: number; item: LineupItem }) => {
```

Comment:
not sure if it makes sense to make this its own component that can dispatch on the item type? could be helpful for reducing complexity of the page. also a potential idea is to pass the render methods into the lineup, and the lineup's core logic is managing pagination etc?